npm test fails and gives errors

Hello all,

Please forgive if I do this wrong. I am fairly noob but not completely. Oh, one other thing (2 issues in one post?) first off the GitHub CLI method did not work for me, the error was;

`peter@pop-os:~/challenge-bypass-extension$ gh repo clone privacypass/challenge-bypass-extension
Cloning into 'challenge-bypass-extension'...
warning: templates not found /usr/share/git-core/templates
fatal: Unable to find remote helper for 'https'
exit status 128`

git clone worked for me though. So the actual issue that I was posting here about is npm test. I got no errors through the build everything seems as it installed correctly when I try to test with "npm test" here is what I get;

`> privacy-pass@3.0.2 pretest
> npm run sjcl


> privacy-pass@3.0.2 sjcl
> cd node_modules/sjcl && perl configure --without-all --with-ecc --with-convenience --with-codecBytes --with-codecHex --compress=none && make sjcl.js

Enabled components:
  aes
  bitArray
  codecString
  codecHex
  codecBase64
  codecBytes
  sha256
  ccm
  hmac
  pbkdf2
  random
  convenience
  bn
  ecc
  exports
  
Compression: none
cat core/sjcl.js core/aes.js core/bitArray.js core/codecString.js core/codecHex.js core/codecBase64.js core/codecBytes.js core/sha256.js core/ccm.js core/hmac.js core/pbkdf2.js core/random.js core/convenience.js core/bn.js core/ecc.js core/exports.js  > core.js
cp core.js sjcl.js

> privacy-pass@3.0.2 test
> node --experimental-vm-modules node_modules/.bin/jest

(node:117964) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
(node:118093) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
(node:118094) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
(node:118092) ExperimentalWarning: VM Modules is an experimental feature. This feature could change at any time
ts-jest[config] (WARN) There is a mismatch between your NodeJs version v12.22.5 and your TypeScript target es2020. This might lead to some unexpected errors when running tests with `ts-jest`. To fix this, you can check https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping
ts-jest[config] (WARN) There is a mismatch between your NodeJs version v12.22.5 and your TypeScript target es2020. This might lead to some unexpected errors when running tests with `ts-jest`. To fix this, you can check https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping
ts-jest[config] (WARN) There is a mismatch between your NodeJs version v12.22.5 and your TypeScript target es2020. This might lead to some unexpected errors when running tests with `ts-jest`. To fix this, you can check https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping
 FAIL  src/background/providers/cloudflare.test.ts
  ● Test suite failed to run

    SyntaxError: The requested module 'crypto' does not provide an export named 'webcrypto'

      at async Runtime.linkAndEvaluateModule (node_modules/jest-runtime/build/index.js:779:5)

 FAIL  src/background/voprf.test.ts
  ● Test suite failed to run

    SyntaxError: The requested module 'crypto' does not provide an export named 'webcrypto'

      at async Runtime.linkAndEvaluateModule (node_modules/jest-runtime/build/index.js:779:5)

 FAIL  src/background/token.test.ts
  ● Test suite failed to run

    SyntaxError: The requested module 'crypto' does not provide an export named 'webcrypto'

      at async Runtime.linkAndEvaluateModule (node_modules/jest-runtime/build/index.js:779:5)

Test Suites: 3 failed, 3 total
Tests:       0 total
Snapshots:   0 total
Time:        2.728 s
Ran all test suites.
npm ERR! code 1
npm ERR! path /home/peter/challenge-bypass-extension
npm ERR! command failed
npm ERR! command sh -c node --experimental-vm-modules node_modules/.bin/jest

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/peter/.npm/_logs/2022-04-15T10_37_07_012Z-debug.log`

So the log file that is referenced follows;

`0 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'test' ]
1 info using npm@7.5.2
2 info using node@v12.22.5
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/usr/share/nodejs/npm/npmrc Completed in 1ms
5 timing config:load:builtin Completed in 1ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 0ms
8 timing config:load:file:/home/peter/challenge-bypass-extension/.npmrc Completed in 0ms
9 timing config:load:project Completed in 1ms
10 timing config:load:file:/home/peter/.npmrc Completed in 0ms
11 timing config:load:user Completed in 0ms
12 timing config:load:file:/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 0ms
15 timing config:load:validate Completed in 0ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 5ms
19 verbose npm-session 4be05afc9ed5b7d7
20 timing npm:load Completed in 11ms
21 timing command:run-script Completed in 3758ms
22 timing command:test Completed in 3759ms
23 verbose stack Error: command failed
23 verbose stack     at ChildProcess.<anonymous> (/usr/share/nodejs/@npmcli/promise-spawn/index.js:64:27)
23 verbose stack     at ChildProcess.emit (events.js:314:20)
23 verbose stack     at maybeClose (internal/child_process.js:1022:16)
23 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
24 verbose pkgid privacy-pass@3.0.2
25 verbose cwd /home/peter/challenge-bypass-extension
26 verbose Linux 5.16.15-76051615-generic
27 verbose argv "/usr/bin/node" "/usr/bin/npm" "test"
28 verbose node v12.22.5
29 verbose npm  v7.5.2
30 error code 1
31 error path /home/peter/challenge-bypass-extension
32 error command failed
33 error command sh -c node --experimental-vm-modules node_modules/.bin/jest
34 verbose exit 1
`

In the error message it gives a URL of `https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping` I pushed ctrl and clicked on it, looked through it and it is beyond my understanding. Just so it is said prior to writing this up I did 2 google searches with a long and short version of the error message but I got nothing specifically with this project and the general results I got were once again, beyond my level of knowledge.

I hope this was clear as well as pertinent.

Oh yes one last thing, don't know if it matters but I did review the contributing guidelines but have no clue what any of that means.

--- a/.npmrc
+++ b/.npmrc
@@ -1,0 +1,1 @@
+engine-strict=true

--- a/package.json
+++ b/package.json
@@ -8,6 +8,9 @@
     "main": "index.js",
     "license": "BSD-3-Clause",
     "type": "module",
+    "engines": {
+        "node": ">=16"
+    },
     "scripts": {
         "sjcl": "cd node_modules/sjcl && perl configure --without-all --with-ecc --with-convenience --with-codecBytes --with-codecHex --compress=none && make sjcl.js",
         "prebuild": "npm run sjcl",

